### PR TITLE
Amex report parser in python

### PR DESCRIPTION
**Pull request by rfroetscher**
Did a super simple port of this over to python

@dollschasingmen @timchan-lumoslabs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## Installation
+`$ pip install pyxmex`
+
+## Usage
+```python
+from pyxmex import Parser
+parser = Parser()
+
+parsed = parser.parse_detail_record_from_file('MYCOMPANY.EPTRN')
```
Comment (dollschasingmen) [inline README.md]: this function name seems a little verbose. i feel the ideas of detail_record and from file should not be explicit in the function name.

Comment (dollschasingmen) [inline README.md]: i feel like i would just call it `process` or something

Comment (rfroetscher) [inline README.md]: fair enough

File: pyxmex/parser.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    def _parse_line(self, section_type, line):
+        fields = {}
+        field_formats = self.eptrn_config['TYPES'][section_type]['FIELDS']
+
+        for field_format in field_formats:
+            fields[field_format['NAME']] = self._range_from_list(line, *field_format['RANGE'])
+
+        return fields
+
+    def left_outer_join_sections(self, left_collection, right_collection, join_condition):
```
Comment (dollschasingmen) [inline pyxmex/parser.py]: this is public interface? maybe write something in the readme on how to use it

**Comment by dollschasingmen** [comment]
looks good man, i made some comments but if this works, it works.

File: pyxmex/joiner.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class Joiner():
+    @classmethod
```
Comment (timchan-lumoslabs) [inline pyxmex/joiner.py]: What is the purpose of the `@classmethod` annotation? It seems like this function is being used like a static method, so why not the `@staticmethod` annotation?

Comment (rfroetscher) [inline pyxmex/joiner.py]: yeah you have it right. i'll switch

**Pull request merged**